DB/src/main/java/io/deephaven/db/v2/parquet/ParquetReaderUtil.java

```diff
@@ -101,6 +104,285 @@ public static void readParquetFile(String filePath) throws IOException {
+                        } else {
+                            colDefConsumer.accept(colName, int.class, null, isGroupinng, codecName, codecArgs);
+                        }
+                        break;
+                    case INT64:
+                        if (isArray) {
+                            colDefConsumer.accept(colName, long[].class, long.class, isGroupinng, codecName, codecArgs);
+                        } else {
+                            colDefConsumer.accept(colName, long.class, null, isGroupinng, codecName, codecArgs);
+                        }
+                        break;
+                    case DOUBLE:
+                        if (isArray) {
+                            colDefConsumer.accept(colName, double[].class, double.class, isGroupinng, codecName, codecArgs);
+                        } else {
+                            colDefConsumer.accept(colName, double.class, null, isGroupinng, codecName, codecArgs);
+                        }
+                        break;
+                    case FLOAT:
+                        if (isArray) {
+                            colDefConsumer.accept(colName, float[].class, float.class, isGroupinng, codecName, codecArgs);
+                        } else {
+                            colDefConsumer.accept(colName, float.class, null, isGroupinng, codecName, codecArgs);
+                        }
+                        break;
+                    case BINARY:
+                    case FIXED_LEN_BYTE_ARRAY:
+                        final String specialType = keyValueMetaData.get(ParquetTableWriter.SPECIAL_TYPE_NAME_PREFIX_ + colName);
+                        final Supplier<String> exceptionTextSupplier = ()
+                                -> "BINARY or FIXED_LEN_BYTE_ARRAY type " + column.getPrimitiveType()
+                                    + " for column " + Arrays.toString(column.getPath());
+                        final Class<?> dbType;
+                        if (specialType != null) {
+                            if (specialType.equals(ParquetTableWriter.STRING_SET_SPECIAL_TYPE)) {
+                                colDefConsumer.accept(colName, StringSet.class, null, isGroupinng, codecName, codecArgs);
+                            } else {
+                                throw new RuntimeException(exceptionTextSupplier.get()
+                                        + " with unknown special type " + specialType);
+                            }
+                        }
+                        throw new RuntimeException(exceptionTextSupplier.get() + " with no special type or encoding metadata.");
+                    default:
+                        throw new RuntimeException("Unsupported type " + column.getPrimitiveType() + " for column " + Arrays.toString(column.getPath()));
+                }
+            } else {
+                final Optional<Class<?>> optionalClass = logicalTypeAnnotation.accept(visitor);
```

Comment:
`.get().orElseThrow(` is probably a better way to use optionals.

Reply:
Disagree that, in this case, that is more readable, giving the string construction we are doing for the exception.
I'll be happy to take your code and put it there if you go through the exercise and after looking at the result you still think is better.

Reply:
Done.